lib/identity_cache/query_api.rb

```diff
@@ -183,6 +184,7 @@ def get_embedded_association(record, association, options) #:nodoc:
 
       def coder_from_record(record) #:nodoc:
         unless record.nil?
+          record.readonly! if IdentityCache.fetch_read_only_records
```

Comment:
Why are we doing this here? Because it's a useful bottleneck interface? (I.e. only all relevant paths go through here.) It seems wrong to modify the `record` argument in a method called `coder_from_record`.